tests/test_report_reader.cpp

```diff
@@ -23,23 +30,89 @@ TEST_CASE("SpikeReader", "[base]") {
             SpikeReader::Population::Sorting::none);
 }
 
-TEST_CASE("ReportReader", "[base]") {
+TEST_CASE("SomaReportReader limits", "[base]") {
     const SomaReportReader reader("./data/somas.h5");
 
     REQUIRE(reader.getPopulationsNames() == std::vector<std::string>{"All", "soma1", "soma2"});
 
-    REQUIRE(reader.openPopulation("All").getTimes() == std::make_tuple(0., 1., 0.1));
+    auto pop = reader.openPopulation("All");
+
+    // ids out of range
+    REQUIRE(pop.get(Selection({{100, 101}})).ids == DataFrame<NodeID>::DataType());
+    testTimes(pop.get(Selection({{100, 101}})).times, 0.0, 0.1, 10);
+    REQUIRE(pop.get(Selection({{100, 101}})).data ==
+            std::vector<std::vector<float>>({{}, {}, {}, {}, {}, {}, {}, {}, {}, {}}));
+
+    // Inverted id
+    REQUIRE_THROWS(pop.get(Selection({{2, 1}})));
+
+    // Negative ids
+    REQUIRE_THROWS(pop.get(Selection({{-1, 1}})));
+
+    // Times out of range
+    REQUIRE(pop.get(Selection({{1, 2}}), 100., 101.).ids == DataFrame<NodeID>::DataType());
+    testTimes(pop.get(Selection({{1, 2}}), 100., 101.).times, 0.0, 0.0, 0);
+    REQUIRE(pop.get(Selection({{1, 2}}), 100., 101.).data == std::vector<std::vector<float>>({}));
+
+    // Inverted times
+    REQUIRE(pop.get(Selection({{1, 2}}), 0.2, 0.1).ids == DataFrame<NodeID>::DataType({1}));
+    testTimes(pop.get(Selection({{1, 2}}), 0.2, 0.1).times, 0.2, 0.1, 1);
+    REQUIRE(pop.get(Selection({{1, 2}}), 0.2, 0.1).data ==
+            std::vector<std::vector<float>>({{1.2}}));
+
+    // Negatives times
+    REQUIRE(pop.get(Selection({{1, 2}}), -1., -2.).ids == DataFrame<NodeID>::DataType({1}));
+    testTimes(pop.get(Selection({{1, 2}}), -1., -2.).times, 0.0, 0.1, 10);
+    REQUIRE(pop.get(Selection({{1, 2}}), -1., -2.).data ==
+            std::vector<std::vector<float>>(
+                {{1.0}, {1.1}, {1.2}, {1.3}, {1.4}, {1.5}, {1.6}, {1.7}, {1.8}, {1.9}}));
```

Comment:
not sure understand why we have values here

Reply:
it means everything

Reply:
ok good. 
Then maybe having tests for : 
```
pop.get(Selection({{1, 2}}), -1., something)

pop.get(Selection({{1, 2}}), 2., -1)
```
can be great.

Reply:
now it throws